searchlib/src/tests/common/bitvector/bitvector_test.cpp

```diff
@@ -844,4 +844,56 @@ TEST(BitvectorTest, fixup_count_and_guard_bit_and_zero_remaining_data_bits_after
     EXPECT_TRUE(bv == bv2);
 }
 
+TEST(BitvectorTest, normal_guard_bits)
+{
+    for (uint32_t i = 0; i < 2; ++i) {
+        auto bv_size = 2048u - BitVector::num_guard_bits - i;
+        SCOPED_TRACE("bv_size=" + std::to_string(bv_size));
+        AllocatedBitVector bv(bv_size);
+        EXPECT_EQ(bv_size + i, bv.capacity());
+        bv.clearInterval(0, bv_size);
+        EXPECT_EQ(bv_size, bv.getFirstTrueBit(0));
+        bv.setInterval(0, bv_size);;
+        if (BitVector::num_guard_bits > 1) {
+            EXPECT_EQ(bv_size + 1, bv.getFirstFalseBit(0));
+        }
+    }
+}
+
+TEST(BitvectorTest, dynamic_guard_bits)
+{
+    vespalib::GenerationHolder g;
+    for (uint32_t i = 0; i < 2; ++i) {
+        auto bv_size = 2048u - BitVector::num_guard_bits - i;
+        constexpr bool sgb = (BitVector::num_guard_bits == 1);
+        constexpr uint32_t slack = sgb ? 0 : 1;
+        SCOPED_TRACE("bv_size=" + std::to_string(bv_size));
+        GrowableBitVector bv(bv_size, bv_size, g, nullptr);
+        EXPECT_EQ(bv_size + i, bv.writer().capacity());
+        bv.writer().clearInterval(0, bv_size);
+        EXPECT_EQ(bv_size + ((i == 0) ? 0 : slack), bv.reader().getFirstTrueBit(0));
+        bv.shrink(257);
+        EXPECT_EQ(257 + slack, bv.reader().getFirstTrueBit(0));
+        bv.shrink(256);
+        EXPECT_EQ(256, bv.reader().getFirstTrueBit(0));
+        bv.shrink(255);
+        EXPECT_EQ(255 + slack, bv.reader().getFirstTrueBit(0));
```

Comment:
I think these checks would benefit from an initial comment briefly explaining the relation with odd/even guard bit positions